generatorreceiver/internal/topology/tag_set.go

```diff
@@ -11,3 +13,25 @@ type TagSet struct {
 	Inherit             []string       `json:"inherit,omitempty" yaml:"inherit,omitempty"`
 	flags.EmbeddedFlags `json:",inline" yaml:",inline"`
 }
+
+func pickBasedOnWeight(tagSets []TagSet) TagSet {
```

Comment:
I moved `pickBasedOnWeight()` from `trace_generator.go` to `tag_set.go`. Not sure if it belongs here (since the function is specific to tagsets) or in `service_tier.go` (since that's the only place the function is currently called from) or elsewhere.

Reply:
I think here makes sense for now, although we will want to genericize this function at some point - `ResourceAttributeSet` has more or less the right shape to work with a generic version of this function, and we have a TODO to do support weight there. If you want to tackle making this generic, great! If not, we'll pick it up down the line. Totally up to you.

If you do decide to dig in, the gist of it is that we'd add an interface, something like "Pickable":
```go
type Pickable interface {
    GetWeight() float64
    ShouldGenerate() float64
}
```

then change `pickBasedOnWeight` to look like
```go
func pickBasedOnWeight[P pickable](ps []P) P {
...
}
```

and add the `GetWeight` method to `TagSet` and `ResourceAttributeSet` - or even make an `EmbeddedWeight` type like `EmbeddedFlags` and put `GetWeight` there.

Reply:
TIL about generics in Go! Updated, let me know how it looks.